src/app/qgisapp.cpp

```diff
@@ -12854,31 +12856,36 @@ bool QgisApp::checkMemoryLayers()
     }
     else if ( it.value() && it.value()->isTemporary() )
       hasTemporaryLayers = true;
+    else if ( it.value() && it.value()->type() == QgsMapLayerType::MeshLayer )
```

Comment:
Instead of this, you should override QgsMapLayer::isTemporary

Reply:
done, but, here I prefer keep the distinguish between mesh layer and other type because the logic is not exactly the same : for other, all the mesh is temporary, for mesh layer, only some dataset groups.